onnxruntime/core/providers/snpe/snpe_lib.cc

```diff
@@ -0,0 +1,401 @@
+                               const std::vector<int64_t>& input_sizes,
+                               const SnpeRuntimeOptions& snpe_settings) {
+  ORT_RETURN_IF_NOT(snpe_settings.GetRuntimeTarget().IsAvailable(),
+                    "Provided runtime target is not available: ",
+                    snpe_settings.GetRuntimeTarget().ToString());
+
+  ORT_RETURN_IF(0 == output_tensor_names.size() || 0 == input_tensor_names.size(),
+                "Input names or output names are empty!");
+
+  zdl::SNPE::SNPEBuilder snpe_builder(container);
+  zdl::DlSystem::StringList dl_output_tensor_names = zdl::DlSystem::StringList();
+  for (auto it = output_tensor_names.begin(); it != output_tensor_names.end(); ++it) {
+    dl_output_tensor_names.append((*it).c_str());
+  }
+
+  buffer_type_ = snpe_settings.GetBufferType();
+  bool use_user_buffer = buffer_type_ == BufferType::ITENSOR ? false : true;
+  snpe_builder.setOutputTensors(dl_output_tensor_names)
+              .setRuntimeProcessor(snpe_settings.GetRuntimeTarget().Get())
+              .setExecutionPriorityHint(snpe_settings.GetExecutionPriority())
+              .setUseUserSuppliedBuffers(use_user_buffer);
+#ifdef __ANDROID__
+  // use sustained performance mode on android variants.
+  LOGS_DEFAULT(INFO) << "setPerformanceProfile to SUSTAINED_HIGH_PERFORMANCE for Android environment!";
+  snpe_builder.setPerformanceProfile(zdl::DlSystem::PerformanceProfile_t::SUSTAINED_HIGH_PERFORMANCE);
+#endif
+
+#ifdef _WIN32
+  if (snpe_settings.GetRuntimeTarget().Get() == zdl::DlSystem::Runtime_t::DSP) {
+    zdl::DlSystem::PlatformConfig platformConfig;
+    ORT_RETURN_IF_NOT(platformConfig.setPlatformOptionValue("unsignedPD", "ON"),
+                      "unsignedPD is not available! ", GetSnpeErrorString());
+    snpe_builder.setPlatformConfig(platformConfig);
+  }
+#endif
+
+  snpe_ = snpe_builder.build();
+  ORT_ENFORCE(snpe_, "Failed to create snpe instance! ", GetSnpeErrorString());
+
+  // Make sure shape of snpe inputs are same with onnx model inputs
+  ORT_RETURN_IF_ERROR(CheckInputsSize(input_tensor_names, input_sizes));
+
+  ORT_ENFORCE(BufferType::UNKNOWN != buffer_type_, "Buffer type unknown!");
+
+  if (BufferType::ITENSOR == buffer_type_) {
+    ORT_RETURN_IF_ERROR(SetupInputTensors(input_tensor_names));
+  } else {
+    ORT_RETURN_IF_ERROR(SetupUserBufferAttributes(input_tensor_names));
+    ORT_RETURN_IF_ERROR(SetupUserBufferAttributes(output_tensor_names));
+  }
+
+  return Status::OK();
+}
+
+Status SnpeLib::Initialize(const char* dlcPath,
+                           const std::vector<std::string>& output_layer_names,
+                           const std::vector<std::string>& input_layer_names,
+                           const std::vector<int64_t>& input_sizes,
+                           const SnpeRuntimeOptions& snpe_settings) {
+  auto container = zdl::DlContainer::IDlContainer::open(zdl::DlSystem::String(dlcPath));
```

Comment:
If I understand it correctly, here it dynamically loads a DLL(or *.so). Then when will it be unloaded?

Reply:
container is unique_ptr